File: python/python_bandwidth_bxml/verbs/forward.py
"""
forward.py

Representation of Bandwidth's forward BXML verb

@author Jacob Mulford
@copyright Bandwidth INC
"""

from lxml import etree

from .base_verb import AbstractBxmlVerb

FORWARD_TAG = "Forward"


class Forward(AbstractBxmlVerb):

    def __init__(self, to=None, from_=None, call_timeout=None, diversion_treatment=None, diversion_reason=None):
        """
        Initializes the Forward class with the following parameters

        :param str to: The phone number destination of the call
        :param str from_: The phone number that the recipient will receive the call from
        :param int call_timeout: The number of seconds to wait before timing out the call
        :param str diversion_treatment: The diversion treatment for the call
        :param str diversion_reason: The diversion reason for the call
        """
        self.to = to
        self.from_ = from_
        self.call_timeout = str(call_timeout)
        self.diversion_treatment = diversion_treatment
        self.diversion_reason = diversion_reason

    def to_xml(self):
        root = etree.Element(FORWARD_TAG, to=self.to, callTimeout=self.call_timeout,
                            diversionTreatment=self.diversion_treatment, diversionReason=self.diversion_reason)
        root.set("from", self.from_) # Workaround since `from` is a keyword in Python and needs to be declared as an attribute separately
        return etree.tostring(root).decode()

File: php/BxmlVerbs/verbs/forward.php
<?php
/**
 * forward.php
 *
 * Implementation of the BXML Forward verb
 *
 * @author Jacob Mulford
 * @copyright Bandwidth INC
 */

namespace BxmlVerbs;

require_once "verb.php";

class Forward extends Verb {

    /**
     * Sets the to attribute for Forward
     *
     * @param string $to The phone number to receive the phone call 
     */
    public function to($to) {
        $this->to = $to;
    }

    /**
     * Sets the from attribute for PlayAudio
     *
     * @param string $from The phone number to make the phone call
     */
    public function from($from) {
        $this->from = $from;
    }

    /**
     * Sets the callTimeout attribute for PlayAudio
     *
     * @param string $callTimeout The timeout in seconds for the phone call 
     */
    public function callTimeout($callTimeout) {
        $this->callTimeout = $callTimeout;
    }

    /**
     * Sets the diversionTreatment attribute for PlayAudio
     *
     * @param string $diversionTreatment The diversion treatment for the phone call 
     */
    public function diversionTreatment($diversionTreatment) {
        $this->diversionTreatment = $diversionTreatment;
    }

    /**
     * Sets the diversionReason attribute for PlayAudio
     *
     * @param string $diversionReason The diversion treatment for the phone call 
     */
    public function diversionReason($diversionReason) {
        $this->diversionReason = $diversionReason;
    }

    public function toXml($doc) {
        $element = $doc->createElement("Forward");

        if(isset($this->to)) {
            $element->setAttribute("to", $this->to);
        }

        if(isset($this->from)) {
            $element->setAttribute("from", $this->from);
        }

        if(isset($this->callTimeout)) {
            $element->setAttribute("callTimeout", $this->callTimeout);
        }

        if(isset($this->diversionTreatment)) {
            $element->setAttribute("diversionTreatment", $this->diversionTreatment);
        }

        if(isset($this->diversionReason)) {
            $element->setAttribute("diversionReason", $this->diversionReason);
        }

        return $element;
    }
}
